enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,647 @@
+
+## Summary
+
+The Assisted Installer currently runs as a SaaS on cloud.redhat.com, enabling
+users to deploy OpenShift clusters with certain customizations, particularly on
+bare metal hardware. It is necessary to bring those capabilities on-premise in
+users' "Hub" clusters by installing clusters via Multi-cluster management, such
+as through [Hive](https://github.com/openshift/hive/) and RHACM (Red Hat
+Advanced Cluster Management).
+
+This enhancement proposes to add the agent-based installation workflow to Hive.
+
+## Motivation
+
+The Assisted Installer running as a SaaS has demonstrated a new and valuable
+way to install OpenShift clusters through an agent-based flow. The [Assisted
+Installer for Connected Environments](connected-assisted-installer.md)
+enhancement proposal elaborates on the value of the agent-based installations.
+
+However many customers do not want a third party service running outside of
+their network to be able to provision hosts inside their network. Reasons
+include trust, control, and reproducibility. Those users will prefer to run the
+service on-premises. There are also many customers whose network environments
+have limited or no connectivity to the internet.
+
+[Hive](https://github.com/openshift/hive/) is the kubernetes-native API for
+installing OpenShift, and it is the natural place to introduce an on-premises
+capability for agent-based cluster installation.
+
+### Goals
+
+* Expose agent-based installation capabilities as a kubernetes-native API
+through [Hive](https://github.com/openshift/hive).
+* Enable using agent-based installation capabilities to add workers to clusters
+that have been installed using - or adopted by - Hive.
+* Enable automated creation and booting of the Assisted discovery ISO for bare-metal deployments.
+* Ensure the design is extensible for installing on other platforms.
+
+### Non-Goals
+
+* Remote Worker Node support requires solving specific problems that will be
+addressed in a separate proposal. For example, it requires the ability to
+ensure that the Node joins with an appropriate role such that it will not be
+schedulable for specific workloads, such as the ingress controller.
+* Solve central machine management. ("central machine management" involves
+running machine-api-providers on a hub cluster in order to manage Machines on
+the hub which represent Nodes in spoke clusters.) While this is a foundational
+element to solving central machine management, this enhancement doesn't attempt
+to address the topic in its totality.
+* Run metal3 components on a non-baremetal cluster. Some parts of this proposal
+assume that the baremetal-operator is present on the hub cluster. Today that is
+only possible when the hub cluster itself is on the bare metal platform. Future
+work may propose running baremetal-operator on non-baremetal hub clusters.
+* Applying static network configuration to hosts. This is a desired feature
```

Comment:
not just static network configuration. We need:
- vlan
- bonding
- capability of disabling certain nics (to avoid dhcp timeouts)

Reply:
Correct, this is all out of scope of this enhancement.